inventories/broker.py

```
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import yaml
+import os
+import subprocess
+import sys
+from io import StringIO
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Broker inventory script")
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--list', action='store_true')
+    group.add_argument('--host')
+    return parser.parse_args()
+
+
+def get_running_hosts():
+    cmd = ["broker", "inventory", "--details"]
+
+    try:
+        output = subprocess.check_output(cmd, universal_newlines=True).rstrip()
+    except FileNotFoundError:
+        return
```

Perhaps raise a `SystemExit('Broker command not found')` instead of silently accepting the error?

---

Since this inventory is optional, I think it's better to "fail" silently. I realize that doesn't help if you intentionally want broker but screwed something up. If this raises an error, you end up with this output everytime:

```
[WARNING]:  * Failed to parse /home/ehelms/workspace/upstream/forklift/inventories/broker.py with script plugin: Inventory script (/home/ehelms/workspace/upstream/forklift/inventories/broker.py) had an execution error: Broker not
found
[WARNING]:  * Failed to parse /home/ehelms/workspace/upstream/forklift/inventories/broker.py with ini plugin: /home/ehelms/workspace/upstream/forklift/inventories/broker.py:3: Expected key=value host variable assignment, got:
argparse
[WARNING]: Unable to parse /home/ehelms/workspace/upstream/forklift/inventories/broker.py as an inventory source
```

We do the same thing for the vagrant.py plugin, for the same reason (you can use forklift without Vagrant). https://github.com/theforeman/forklift/blob/master/inventories/vagrant.py#L36-L39

---

If Ansible always loads all inventories in `inventories` then I think you're right.